test/common/stats/symbol_table_impl_test.cc

```diff
@@ -180,5 +182,50 @@ TEST_F(StatNameTest, TestShrinkingExpectation) {
   EXPECT_EQ(table_size_0, table_.size());
 }
 
+#ifdef ENABLE_MEMORY_USAGE_TESTS
+
+// Tests the memory savings realized from using symbol tables with 1k clusters. This
+// test shows the memory drops from almost 8M to less than 2M.
+TEST(SymbolTableTest, Memory) {
+  // Tests a stat-name allocation strategy.
+  auto test_memory_usage = [](std::function<void(absl::string_view)> fn) -> size_t {
+    size_t start_mem = Memory::Stats::totalCurrentlyAllocated();
+    TestUtil::foreachStat(1000, fn);
+    size_t end_mem = Memory::Stats::totalCurrentlyAllocated();
+    if (end_mem != 0) { // See warning below for asan, tsan, and mac.
+      EXPECT_GT(end_mem, start_mem);
+    }
+    return end_mem - start_mem;
+  };
+
+  size_t string_mem_used, symbol_table_mem_used;
+  {
+    std::vector<std::string> names;
+    auto record_stat = [&names](absl::string_view stat) { names.push_back(std::string(stat)); };
+    string_mem_used = test_memory_usage(record_stat);
+  }
+  {
+    SymbolTableImpl table;
+    std::vector<StatNamePtr> names;
+    auto record_stat = [&names, &table](absl::string_view stat) {
+      names.emplace_back(table.encode(stat));
+    };
+    symbol_table_mem_used = test_memory_usage(record_stat);
+  }
+
+  // This test only works if Memory::Stats::totalCurrentlyAllocated() works, which
+  // appears not to be the case in some tests, including asan, tsan, and mac.
+  if (Memory::Stats::totalCurrentlyAllocated() == 0) {
```

Comment:
Maybe not...actually I think I'll clean this up. There are now two tests:
 1) are we using tcmalloc
 2) does the malloc-stats library return anything interesting.

I think I'll just make it one functional test so that (a) we always compile the tests and (b) we never run them if they don't do anything.